source/components/button-upload/button-upload-component.js

```diff
@@ -36,12 +37,25 @@ class ButtonUpload extends PureComponent {
     btnText: 'Upload',
     disabled: false,
     limit: false,
+    type: 'default',
+    files: [],
     loading: false,
     formatWhiteList: ['.png', '.jpg', '.jpeg', '.pdf'],
     maxFileSize: 3000000, // 3MB
     allowedDimensions: [],
   };
 
+  componentDidUpdate(prevProps) {
+    if (this.props.type === 'controlled') {
+      const { files } = this.props;
```

Comment:
Aí se for seguir a sugestão de só usar o defaultFiles, seria só trocar esses files aqui por defaultFiles